utility/src/main/java/org/oppia/util/data/UrlImageParser.kt

```diff
@@ -52,10 +49,16 @@ class UrlImageParser(
         val textviewWidth = tvContents.width
         val drawableHeight = (drawable as BitmapDrawable).intrinsicHeight
         val drawableWidth = (drawable as BitmapDrawable).intrinsicWidth
-        val newHeight = drawableHeight * textviewWidth / drawableWidth
-        val rect = Rect(0, 0, textviewWidth, newHeight)
-        (drawable as BitmapDrawable).bounds = rect
-        urlDrawable.bounds = rect
+        if (drawableWidth > textviewWidth) {
+          val calculatedHeight = textviewWidth * drawableHeight / drawableWidth;
+          val rect = Rect(0, 0, textviewWidth, calculatedHeight)
+          (drawable as BitmapDrawable).bounds = rect
+          urlDrawable.bounds = rect
+        } else {
+          val rect = Rect(0, 0, drawableWidth, drawableHeight)
+          (drawable as BitmapDrawable).bounds = rect
+          urlDrawable.bounds = rect
+        }
```

Comment:
Please give KDoc for the inner class and the logic applied

Reply:
added KDoc